React/Profiler/RCTProfile.m

```diff
@@ -42,8 +42,7 @@
 
 #pragma mark - Variables
 
-// This is actually a BOOL - but has to be compatible with OSAtomic
-static volatile uint32_t RCTProfileProfiling;
+static atomic_bool RCTProfileProfiling = ATOMIC_VAR_INIT(NO);
```

Comment:
Can you mark this as `constexpr` so we can verify at compile-time this doesn't add a global initalizer?